lib/iris/fileformats/um/_fast_load_structured_fields.py

```diff
@@ -88,11 +87,11 @@ def data(self):
         if not self._structure_calculated:
             self._calculate_structure()
         if self._data_cache is None:
-            data_arrays = np.array([as_lazy_data(f._data, chunks=f._data.shape)
-                for f in self.fields], dtype=object)
-            data_arrays = data_arrays.reshape(
-                self.vector_dims_shape + data_arrays.shape[1:])
-            self._data_cache = multidim_daskstack(data_arrays)            
+            stack = np.empty(np.prod(self.vector_dims_shape), 'object')
+            for index, f in enumerate(self.fields):
```

Comment:
Good point. I used f as that was what was used in the previous code. I will change it to 
`for index, field in ...`